mkosi/__init__.py

```diff
@@ -2586,10 +2638,13 @@ def install_debian_or_ubuntu(args: MkosiArgs, root: Path, *, do_run_build_script
         # Debian still has pam_securetty module enabled, disable it in the base image.
         disable_pam_securetty(root)
 
-    if args.distribution == Distribution.debian and "systemd" in extra_packages:
+    if (args.distribution == Distribution.debian and "systemd" in extra_packages and
+            ("systemd-resolved" not in extra_packages)):
```

Comment:
Why would we not want to fixup resolv.conf if systemd-resolved is in the package list?

Reply:
because it is done automatically by the package (it's the reason I added a separate package for it, to takeover automatically)